src/game.rs

```diff
@@ -3092,6 +3203,167 @@ mod tests {
+                BetSizeCandidates::try_from(("50%", "")).unwrap(),
+                Default::default(),
+            ],
+            ..Default::default()
+        };
+
+        let action_tree = ActionTree::new(tree_config).unwrap();
+        let mut game = PostFlopGame::with_config(card_config, action_tree).unwrap();
+
+        let lines = vec![vec![
+            Action::Check,
+            Action::Check,
+            Action::Chance(2),
+            Action::Bet(30),
+        ]];
+
+        let res = game.remove_lines(&lines);
+        assert!(res.is_ok());
+
+        game.allocate_memory(false);
+
+        // Check that the removed line
+        game.back_to_root();
+        game.play(game.node().actions.binary_search(&Action::Check).unwrap());
+        game.play(game.node().actions.binary_search(&Action::Check).unwrap());
+        game.play(2);
+        assert!(game.available_actions() == &[Action::Check]);
+        assert!(game.node().children.len() == 1);
+
+        // Check that other lines are correct
+        game.back_to_root();
+        game.play(game.node().actions.binary_search(&Action::Check).unwrap());
+        game.play(game.node().actions.binary_search(&Action::Check).unwrap());
+        game.play(3);
+        assert!(game.available_actions() == &[Action::Check, Action::Bet(30)]);
+        assert!(game.node().children.len() == 2);
+
+        solve(&mut game, 10, 0.05, true);
+    }
+
+    #[test]
+    fn remove_lines2() {
+        use crate::bet_size::BetSizeCandidates;
+        let card_config = CardConfig {
+            range: ["TT+,AKo,AQs+".parse().unwrap(), "AA".parse().unwrap()],
+            flop: flop_from_str("2c6dTh").unwrap(),
+            ..Default::default()
+        };
+
+        // Simple tree: force checks until river, where OOP can bet 1/2 pot
+        let tree_config = TreeConfig {
+            starting_pot: 60,
+            effective_stack: 970,
+            turn_bet_sizes: [
+                BetSizeCandidates::try_from(("50%", "")).unwrap(),
+                Default::default(),
+            ],
+            river_bet_sizes: [
+                BetSizeCandidates::try_from(("50%", "")).unwrap(),
+                Default::default(),
+            ],
+            ..Default::default()
+        };
+
+        let action_tree = ActionTree::new(tree_config).unwrap();
+        let mut game = PostFlopGame::with_config(card_config, action_tree).unwrap();
+
+        let lines = vec![
+            vec![
+                Action::Check,
+                Action::Check,
+                Action::Chance(2),
+                Action::Bet(30),
+            ],
+            vec![
+                Action::Check,
+                Action::Check,
+                Action::Chance(2),
+                Action::Check,
+                Action::Check,
+                Action::Chance(3),
+                Action::Bet(30),
+            ],
+        ];
+
+        let res = game.remove_lines(&lines);
+        println!("result: {res:?}");
```

Comment:
I prefer not to use the `println!` macro in normal tests.

Reply:
good catch, thought i got rid of all the printing